packages/plugins/persisted-operations/src/index.ts

```diff
@@ -94,6 +94,19 @@ export type CustomPersistedQueryErrors = {
   keyNotFound?: CustomErrorFactory;
 };
 
+const isPersistedOperationContextSymbol = Symbol.for('hive_is_persisted_operation_context');
+
+/**
+ * Helper function for determining whether the execution is using a persisted document.
+ */
+export function isPersistedOperationContext<TContext extends object>(context: TContext): boolean {
+  return isPersistedOperationContextSymbol in context;
+}
+
+function markContextAsPersistedOperationContext<TContext extends object>(context: TContext) {
+  (context as Record<string | symbol, unknown>)[isPersistedOperationContextSymbol] = true;
```

Comment:
Can you use a WeakMap with the context as key to mark it as persisted operation ?

Reply:
WeakMaps can be tricky between different plugins because one plugin will need to depend on another in this case, also might not work properly when versions are different etc. Symbol prop seems safer imo.